cpp/src/arrow/python/python_to_arrow.cc

```diff
@@ -650,6 +652,19 @@ class PyListConverter : public ListConverter<T, PyConverter, PyConverterTrait> {
     return this->value_converter_->Extend(value, size);
   }
 
+  Status AppendIterable(PyObject* value) {
+    int64_t current_size = 0;
+    PyObject* iterator = PyObject_GetIter(value);
+    OwnedRef iter_ref(iterator);
+    while (PyObject* item = PyIter_Next(iterator)) {
+      current_size += 1;
+      OwnedRef item_ref(item);
+      RETURN_NOT_OK(this->value_converter_->Reserve(current_size));
```

Comment:
Are you sure you are using this right? It seems that `Reserve(current_size)` would reserve space for `current_size` elements _in addition_ to the elements already present.

Reply:
You were right, the argument of `Reserve` is the additional capacity to add `Status Reserve(int64_t additional_capacity)`. Changed this to avoid reserving unecessary space.